packages/components/nodes/vectorstores/Postgres/Postgres.ts

```diff
@@ -253,7 +253,8 @@ const similaritySearchVectorWithScore = async (
         port: postgresConnectionOptions.port,
         user: postgresConnectionOptions.username,
         password: postgresConnectionOptions.password,
-        database: postgresConnectionOptions.database
+        database: postgresConnectionOptions.database,
+        ssl: postgresConnectionOptions.extra?.ssl
```

Comment:
Can we use the ... additionalConfig? Instead of just ssl

Reply:
@HenryHengZJ , so `additionalConfiguration` isn't accessible within the `similaritySearchVectorWithScore` function, because it was never passed `nodeData`.

I understand that you're trying to make `poolOptions` appear similar in function to `postgresConnectionOptions`, but it is hard to recreate that because the function doesn't have all the inputs it needs to access `additionalConfiguration`.

I'm open to an alternative solution here, but could you give me a clearer idea of what you're proposing instead?

Reply:
One alternative would be to structure the `poolOptions` declaration like this, instead:

```
const poolOptions = {  
    ...postgresConnectionOptions,  
    host: postgresConnectionOptions.host,  
    port: postgresConnectionOptions.port,  
    user: postgresConnectionOptions.username,  
    password: postgresConnectionOptions.password,  
    database: postgresConnectionOptions.database,  
    ssl: postgresConnectionOptions.extra?.ssl || postgresConnectionOptions.ssl,  
}  
```

^ @HenryHengZJ , does this work for you and satisfy your original intent?

Reply:
hmm im not sure if this works, but can we just get rid of the `poolOptions` and just do `const pool = new Pool(postgresConnectionOptions)` ? because `postgresConnectionOptions` already contains all the configs, so we don't have to redeclare the options again with `poolOptions`

Reply:
@HenryHengZJ , sure, but there's one catch with that approach: `pg` Pool expects all the `ssl` settings to be nested within the `ssl` key and not under `extra.ssl`.  Here's some example code that illustrates how to do this:

```
let postgresConnectionOptions = {  
    host: 'localhost',  
    port: 5432,  
    user: 'youruser',  
    password: 'yourpassword',  
    database: 'yourdatabase',  
    ssl: true,  
    extra: {  
        ssl: {  
            rejectUnauthorized: false,  
            ca: "<your_self_signed_cert_name>"  
        }  
    }  
}  
  
if (postgresConnectionOptions.extra?.ssl) {  
    postgresConnectionOptions.ssl = {  
        ...postgresConnectionOptions.ssl,  
        ...postgresConnectionOptions.extra.ssl  
    };  
    delete postgresConnectionOptions.extra;  
}  
  
const pool = new Pool(postgresConnectionOptions);
```

Reply:
How bout we just delete the `sslConnection` and let the user define their own SSL logic in the `additionalConfig` since it is a JSON from user's input: 
![image](https://github.com/FlowiseAI/Flowise/assets/26460777/c755ecaf-0e3f-4dbf-8b55-f8a5c5bb1ace)

Reply:
@HenryHengZJ , that works.  To be crystal clear, would the new logic expect `additionalConfig` to look like:

1)
Keeping the `extra` field (where we still have to deal with cleaning this up before passing it through to `pg`):
```
{  
  "ssl": true,  
  "extra": {  
    "ssl": {  
      "rejectUnauthorized": false,  
      "ca": "<your_self_signed_cert_name>"  
    }  
  }  
}  
```

2) 
OR would it look like this, in not keeping the `extra` field (and we pass it straight through to `pg`):
```
{  
  "ssl": true  
} 

{  
  "ssl": {  
    "rejectUnauthorized": false,  
    "ca": "<your_self_signed_cert_name>"  
  }  
}   
```

What are you thinking here, @HenryHengZJ ?

Reply:
it will be up to user responsibility to add the `extra` field in the additionalConfig when needed. The SSL thing is tricky, some works with extra, some doesnt, some with boolean, some with cert. So the best way is to let user specify on their own and we just parsed the JSON

Reply:
@HenryHengZJ , okay, sounds good.  Shall I proceed with the refactor on this PR, then?

All of these changes will be Postgres-specific and not apply to any other vector stores, correct?

Reply:
yep!